packages/mui-private-theming/src/ThemeProvider/ThemeProvider.js

```diff
@@ -52,7 +52,8 @@ function ThemeProvider(props) {
   }
 
   const theme = React.useMemo(() => {
-    const output = outerTheme === null ? localTheme : mergeOuterLocalTheme(outerTheme, localTheme);
+    const output =
+      outerTheme === null ? { ...localTheme } : mergeOuterLocalTheme(outerTheme, localTheme);
```

Comment:
this change fixed
`Mutating component props or hook arguments is not allowed. Consider using a local variable insteadeslint(react-compiler/react-compiler)` issue [here](https://github.com/mui/material-ui/pull/42619/files#diff-0b19f9c894672081e57b40704958d8339ad1ac2696888c1f1220eb1720913501R59)